tests/Doctrine/Tests/DBAL/Functional/Ticket/DBAL3516Test.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Ticket;
+
+use Doctrine\Tests\DbalFunctionalTestCase;
+
+/**
+ * @group DBAL-3516
+ */
+class DBAL3516Test extends DbalFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $platform = $this->connection->getDatabasePlatform()->getName();
+
+        if ($platform !== 'mssql') {
```

Comment:
Since this is not the only driver/platform which implements `lastInsertId()`, the test should be implemented for all supported platforms and skipped if there's a known issue with a specific platform.

Reply:
I thought about make this tests for all platforms, but there is a problem with create trigger for all platforms. So I think that creating it only for MSSQL is ok, because this problem probably doesn't concern others platforms.